spi/contract-spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/contract/negotiation/command/ContractNegotiationCommand.java

```diff
@@ -0,0 +1,31 @@
+/*
+ *  Copyright (c) 2021-2022 Fraunhofer Institute for Software and Systems Engineering
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.spi.types.domain.contract.negotiation.command;
+
+import org.eclipse.dataspaceconnector.spi.command.Command;
+
+/**
+ * Sub-type of {@link Command} for modifying ContractNegotiations.
+ */
+public abstract class ContractNegotiationCommand extends Command {
```

Comment:
Maybe I'm missing something, but why do we need the `ContractNegotiationCommand` and the `SingleNegotiationCommand`? Analogously, the `TransferProcessCommand`...

Reply:
I created the `ContractNegotiationCommand` as a super-class for all commands targeting ContractNegotiations. Same for the `TransferProcessCommand` and TransferProcesses. The idea was that these two classes can be used for typing, so e.g. the `TransferProcessManager` has a `CommandQueue<TransferProcessCommand>`, to avoid that a command modifying a ContractNegotiation gets executed within the transfer state machine. The `Single*Commands` always target a specific TransferProcess/ContractNegotiation, whereas another class extending the TransferProcessCommand could also modify multiple TransferProcesses. Does that make sense?